ximea_ros2_cam/src/ximea_ros2_cam.cpp

```diff
@@ -664,6 +694,16 @@ void XimeaROSCam::initParams(){
     this->get_parameter("img_capture_timeout", this->cam_img_cap_timeout_);
     RCLCPP_INFO_STREAM(this->get_logger(),  "cam_img_cap_timeout_: " << this->cam_img_cap_timeout_);
 
+    //      -- apply downsampling parameters --
+    this->declare_parameter( "cam_downsampling", std::string("XI_DWN_1x1"));
+    this->get_parameter("cam_downsampling", this->cam_downsampling_);
+    RCLCPP_INFO_STREAM(this->get_logger(),  "cam_downsampling: " << this->cam_downsampling_);
```

Comment:
I create a macro called LOG_IF to be able to disable by parameter the debug mode (because the debugs was already in the package, and can be helpful considering the high number of camera configs). Commit: https://github.com/Brazilian-Institute-of-Robotics/ximea_ROS2_driver/pull/1/commits/2bb6e0eae9117872aa357f1acf15b0cc064acba2